### PR TITLE
Improve zookeeper

### PR DESCRIPTION
**Pull request by liwagu**
* Support for zookeeper with prometheus monitoring
* Support zookeeper backup and restore

resolve https://github.com/apecloud/kubeblocks/issues/5649

### CODE REVIEW

File: addons/zookeeper/dataprotection/restore.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+set -o errexit
+set -o pipefail
+set -o nounset
+
+DATA_DIR=${DATA_DIR:-/bitnami/zookeeper/data}
+BACKUP_DIR=/data/backup
+
+echo "Restoring Zookeeper data directory..."
+
+LATEST_BACKUP=$(ls -t "${BACKUP_DIR}" | head -1)
```
Comment (nayutah) [inline addons/zookeeper/dataprotection/restore.sh:12]: The backup dir may be empty, so the 'LATEST_BACKUP' is nil

Comment (liwagu) [reply addons/zookeeper/dataprotection/restore.sh:12]: I just hard coded BACKUP_DIR=/data/backup. So what should it be?

File: addons/zookeeper/dataprotection/backup.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+set -o pipefail
+set -o nounset
+
+DATA_DIR=${DATA_DIR:-/bitnami/zookeeper/data}
+BACKUP_DIR=/data/backup
+
+mkdir -p "${BACKUP_DIR}"
+
+echo "Backing up Zookeeper data directory..."
+cp -r "${DATA_DIR}" "${BACKUP_DIR}/zookeeper-data-$(date +%Y%m%d-%H%M%S)"
```
[the remark this answers is not in the document]
Comment (liwagu) [reply addons/zookeeper/dataprotection/backup.sh]: I commit again, please check https://github.com/apecloud/kubeblocks-addons/pull/441/commits/c741b96c35b71d937644b587b3033ab9bee63747

**Comment by liwagu** [comment]
@wangyelei

File: addons/zookeeper/dataprotection/backup.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (wangyelei) [reply addons/zookeeper/dataprotection/backup.sh]: hi @liwagu , seems to have forgotten `Restore` custom resource？

**Comment by wangyelei** [comment]
It seems that the monitor is also not working.

**Pull request closed without merging**